File: src/_lines.ts
import * as fs from "fs";

export const all_lines = () => {
  const run = async () => {
    const file = await fs.readFileSync("lines", "utf-8");
    file.split("\n").forEach((r) => console.log(r));
  };

  run();
};

export const every_other_line = () => {
  const run = async () => {
    const file = await fs.readFileSync("lines", "utf-8");
    // file.split("\n").forEach((r, i) => i % 2 && console.log(r));
    file
      .split("\n")
      .filter((_, i) => i % 2 === 0)
      .forEach((r) => console.log(r));
  };

  run();
};

/*
in order
- every other line
- skip first 2
- print 2
*/
export const custom = () => {
  const run = async () => {
    const file = await fs.readFileSync("lines", "utf-8");
    // file.split("\n").forEach((r, i) => i % 2 && console.log(r));
    file
      .split("\n")
      .filter((_, i) => i % 2 === 0)
      .map((d) => {
        console.log(`debug 1: ${d}`);
        return d;
      })
      .filter((_, i) => i > 1 && i < 4)
      .map((d) => {
        console.log(`debug 2: ${d}`);
        return d;
      })
      .forEach((r) => console.log(r));
  };

  run();
};

File: src/_lines.rs
fn all_lines() {
    let file = std::fs::read_to_string("lines");
    match file {
        Ok(filestr) => {
            // filestr.lines().for_each(|x| println!("{:?}", x)); // also can
            filestr
                .split("\n")
                .into_iter()
                .for_each(|x| println!("{:?}", x));
        }
        Err(msg) => {
            println!("Error: {:?}. Ending Program.", msg.to_string())
        }
    }
}

fn every_other_line() {
    let file = std::fs::read_to_string("lines");
    match file {
        Ok(filestr) => {
            filestr
                .lines()
                .enumerate()
                .filter(|(idx, _)| idx % 2 == 0)
                .for_each(|(_, x)| println!("{}", x));
        }
        Err(msg) => {
            println!("Error: {:?}. Ending Program.", msg.to_string())
        }
    }
}

/*
in order
- every other line
- skip first 2
- print 2
*/
fn custom() {
    let file = std::fs::read_to_string("lines");
    match file {
        Ok(filestr) => {
            filestr
                .lines()
                .enumerate()
                .filter(|(idx, _)| idx % 2 == 0)
                .skip(2)
                .take(2)
                .for_each(|(_, x)| println!("{}", x));
        }
        Err(msg) => {
            println!("Error: {:?}. Ending Program.", msg.to_string())
        }
    }
}
